float64 locationX
float64 locationY
float64 orientation